codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/EndpointResolverGenerator.kt

```diff
@@ -19,155 +24,46 @@ import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
-        hostname.ifPresent {
-            val delimiter = if (protocols.isPresent || credentialScope.isPresent || signatureVersions.isPresent) "," else ""
-            writer.write("hostName: \$S$delimiter", it)
-        }
-
-        protocols.ifPresent {
-            writer.writeInline("protocols: [")
-            val delimiter = if (it.count() == 1) "" else ", "
-            it.forEach {
-                writer.writeInline("\$S$delimiter", it.expectStringNode().value)
-            }
-            val paramDelimiter = if (credentialScope.isPresent || signatureVersions.isPresent) "," else ""
-            writer.write("]$paramDelimiter")
-        }
-
-        credentialScope.ifPresent {
-            writer.writeInline("credentialScope: \$N(", AWSClientRuntimeTypes.Core.CredentialScope)
-            val region = it.getStringMember("region")
-            val service = it.getStringMember("service")
-            region.ifPresent {
-                writer.writeInline("region: \$S", it.value)
-            }
-            if (region.isPresent && service.isPresent) {
-                writer.write(", ")
-            }
-            service.ifPresent {
-                writer.writeInline("serviceId: \$S", it.value)
-            }
-            val paramDelimiter = if (signatureVersions.isPresent) "," else ""
-            writer.write(")$paramDelimiter")
-        }
+    private fun renderResolverProtocol(writer: SwiftWriter) {
+        writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
 
-        signatureVersions.ifPresent {
-            writer.writeInline("signatureVersions: [")
-            val delimiter = if (it.count() == 1) "" else ", "
-            it.forEach { writer.writeInline("\$S$delimiter", it.expectStringNode().value) }
-            writer.write("]")
+        writer.openBlock("public protocol EndpointResolver {", "}") {
+            writer.write("func resolve(params: EndpointParams) throws -> \$L", ClientRuntimeTypes.Core.Endpoint)
         }
     }
 
-    private class PartitionNode(endpointPrefix: String, val config: ObjectNode) {
-        companion object {
-            val comparePartitions = object : Comparator<PartitionNode> {
-                override fun compare(x: PartitionNode, y: PartitionNode): Int {
-                    // always sort standard aws partition first
-                    if (x.id == "aws") return -1
-                    return x.id.compareTo(y.id)
-                }
+    private fun renderResolver(writer: SwiftWriter) {
+        writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
+
+        writer.openBlock("public struct DefaultEndpointResolver: EndpointResolver  {", "}") {
+            writer.write("public init() {}")
+            writer.openBlock(
+                "public func resolve(params: EndpointParams) throws -> \$L {", "}", ClientRuntimeTypes.Core.Endpoint
+            ) {
+                writer.write("fatalError(\"EndpointResolver not implemented\")")
             }
```

Comment:
**Question**: Coming in a future PR? Or does something else replace this to ensure that a default endpoint resovler is available?